### ISSUE TITLE
Cancel upload not working

### ISSUE THREAD

**Issue by NotSoImportant** [commons-app/apps-android-commons#4348]
Not sure what internally happens, but let me describe what I saw:
1. Uploading an image via the share menu from my galery. "File not found error" 
2. Trying to upload the same image via the apps "+" button. See the first uploaded image now. Second attempt only contains part of the description. Clicking the "||" pause button to cancel. Shows message "resume". Upload seems to continue. And afterwards shows both uploads. Second however cannot be found online. 
3. Logging out and in. App now only shows one.

Maybe related to #199 and/or #3960

**Comment by NotSoImportant** [comment]
I was able to reproduce a related behaviour:
- start upload dialog via the share menu of another app
- enter a title
- go back (app crash dialog appeared)
- again share a picture with commons app to get the upload dialog
- this time proceed until upload
- result: at hitting "upload" the same image will be uploaded twice
-

**Comment by 4D17Y4** [comment]
I'd like to work on this one.

**Comment by Rishavgupta12345** [comment]
Yes, pausing/canceling an upload while it is still uploading is not working, we have to add a proper stopping and resuming upload feature 

I would like to work on this issue please assign me @nicolas-raoul

**Comment by nicolas-raoul** [comment]
Oops sorry 4D17Y4 for missing this one!
@Rishavgupta12345 It is yours, thanks :-)

**Comment by cakepineapple** [comment]
I'd like to look into this if this isn't assigned @nicolas-raoul

**Comment by nicolas-raoul** [comment]
@cakepineapple When you have implemented your currently assigned issue, please ask again. Thanks for your interest!
